domain/access/service/user.go

```diff
@@ -20,23 +20,10 @@ import (
 	"github.com/juju/juju/internal/auth"
 )
 
-// UserService provides the API for working with users.
-type UserService struct {
-	st UserState
-}
-
-// NewUserService returns a new UserService for interacting with the underlying user
-// state.
-func NewUserService(st UserState) *UserService {
-	return &UserService{
-		st: st,
-	}
-}
```

Comment:
Why did we make this change in the end?

Reply:
So that we could put the `GetModelUsers` on unified State and Service, since the function makes use of funcitons from each of the Permissions and User state (`authorizedOnTarget` and `checkModelExists`). I could duplicate those functions but this was suggested and it seems like a good idea.

The service is also unified because the states of each service contained a state facade with only the methods from that corresponding state in them. Since `GetModelUsers` was on the direct state object the right thing to do appeared to be unifying the facades and putting `GetModelUsers` on that. 

At this point the User and Permission service are only differentiated by the struct names, I thought that is made more sense just to combine them, I can revert that though if you think its a bad idea.